brax/physics/colliders.py

```diff
@@ -581,11 +620,13 @@ def _collide_pair(config: config_pb2.Config, body_a: bodies.Body,
   impulse_d = math.safe_norm(rel_vel_d) / ((1. / body_a.mass) +
                                            (1. / body_b.mass) + ang)
   # drag magnitude cannot exceed max friction
-  impulse_d = jnp.where(impulse_d < config.friction * impulse, impulse_d,
-                        config.friction * impulse)
+  impulse_d_a = jnp.where(impulse_d < col_a.friction * impulse, impulse_d,
+                        col_a.friction * impulse)
+  impulse_d_b = jnp.where(impulse_d < col_b.friction * impulse, impulse_d,
+                        col_b.friction * impulse)
   dir_d = rel_vel_d / (1e-6 + math.safe_norm(rel_vel_d))
-  dp_d_a = body_a.impulse(qp_a, impulse_d * dir_d, pos_c)
-  dp_d_b = body_b.impulse(qp_b, -impulse_d * dir_d, pos_c)
+  dp_d_a = body_a.impulse(qp_a, impulse_d_a * dir_d, pos_c)
```

Comment:
I believe that a single set of material properties governs how two bodies interact.  The way other engines handle this is they let the user specify a policy for how to combine the properties.  See for example Unity:

https://docs.unity3d.com/Manual/class-PhysicMaterial.html

"When two bodies are in contact, the same bounciness and friction effect is applied to both of them..."

So what is missing then is a policy for how to combine the frictions.  I suggest for now we just include a sane default, which is to multiply the frictions together.  So if one material has 0.6, and another has 1.0, then the resulting friction is 0.6.

In some future PR we can perhaps allow the user to specify how to combine (minimum, max, average, multiply, etc)

Reply:
Ok, will make a shared friction coefficient using the product rule by default.

Reply:
@erikfrey For clarification, should this policy be specific to each body or applied to the entire environment?